### PR TITLE
Filter unwanted Jupyter kernels in Universal image

### PR DESCRIPTION
**Pull request by jungaretti**
Adds the `jupyter.kernels.filter` VS Code setting to the Universal image to filter unwanted Python kernels. This setting makes for a much cleaner VS Code experience when using the Jupyter extension.

This won't filter new Conda environments or Python installations that users add. It just hides the extra installations in the universal image that don't include our Jupyter and ML tooling.

Before:
Many Python installations in the kernel picker

Note that the `Suggested` kernel isn't our primary installation of Python and that there are several entries for our primary installation (because of our symlinks).

After:
Only our primary Python installation in the kernel picker

The only kernel that appears in the picker is our primary installation of Python. Its path matches the output of `which python3`.

### CODE REVIEW

**Review by samruddhikhandale — COMMENTED** [review]
🚀  Looks great to me! 

@jungaretti Could you help bump the patch version in https://github.com/devcontainers/images/blob/main/src/universal/manifest.json#L2 so that we are ready to publish the `universal` image with this change?

File: src/universal/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -111,7 +111,7 @@
                 "go.toolsManagement.checkForUpdates": "local",
                 "go.useLanguageServer": true,
                 "go.gopath": "/usr/local/go/bin/go",
```
Comment (samruddhikhandale) [inline src/universal/.devcontainer/devcontainer.json:113]:
```suggestion
                "go.gopath": "/go",
```

Could you help fix this when you are at it? 😄 
We have been going back and forth with `GOPath` variable, but seems like this should be `/go` - syncing with the GO Feature. 
Reference - https://github.com/devcontainers/features/blob/main/src/go/devcontainer-feature.json#L30

File: src/universal/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -123,6 +123,36 @@
...
+                        "type": "pythonEnvironment"
+                    },
+                    {
+                        "path": "/bin/python3",
+                        "type": "pythonEnvironment"
+                    }
+                ],
                 "lldb.executable": "/usr/bin/lldb",
                 "files.watcherExclude": {
                     "**/target/**": true
```
Comment (samruddhikhandale) [inline src/universal/.devcontainer/devcontainer.json:158]: @Chuxel  could you help me understand the scenario wherein we need to exclude this? 🤔 
Not sure if we should clean this up if it's no longer needed.

**Review by samruddhikhandale — APPROVED** [review]

File: src/universal/.devcontainer/devcontainer.json (hunk 2 above)
Comment (jungaretti) [reply src/universal/.devcontainer/devcontainer.json:158]: https://github.com/devcontainers/images/pull/141

**Review by joshspicer — APPROVED** [review]

**Pull request merged**